src/vtvalidate.cpp

```diff
@@ -118,32 +134,31 @@ NAN_METHOD(isValid) {
     }
     v8::Local<v8::Function> callback = info[1].As<v8::Function>();
 
-    // Check first argument, should be an 'options' object
-    if (!info[0]->IsObject()) {
-        return utils::CallbackError("first arg 'options' must be an object",
-                                    callback);
+    // BUFFER: check first argument, should be a pbf object
+    v8::Local<v8::Value> buffer_val = info[0];
+    if (buffer_val->IsNull() ||
+        buffer_val->IsUndefined()) {
+        utils::CallbackError("first arg is empty", callback);
+        return;
     }
-    v8::Local<v8::Object> options = info[0].As<v8::Object>();
-
-    // Check options object for the "louder" property, which should be a boolean
-    // value
-    if (options->Has(Nan::New("louder").ToLocalChecked())) {
-        v8::Local<v8::Value> louder_val =
-            options->Get(Nan::New("louder").ToLocalChecked());
-        if (!louder_val->IsBoolean()) {
-            return utils::CallbackError("option 'louder' must be a boolean",
-                                        callback);
-        }
-        louder = louder_val->BooleanValue();
+    v8::Local<v8::Object> buffer = buffer_val->ToObject();
+
+    if (buffer->IsNull() ||
+        buffer->IsUndefined() ||
+        !node::Buffer::HasInstance(buffer)) {
+        utils::CallbackError("first arg 'buffer' must be a Protobuf buffer object", callback);
+        return;
     }
 
+    vtzero::vector_tile tile{node::Buffer::Data(buffer), node::Buffer::Length(buffer)};
+
     // Creates a worker instance and queues it to run asynchronously, invoking the
     // callback when done.
     // - Nan::AsyncWorker takes a pointer to a Nan::Callback and deletes the
     // pointer automatically.
     // - Nan::AsyncQueueWorker takes a pointer to a Nan::AsyncWorker and deletes
     // the pointer automatically.
-    auto* worker = new AsyncHelloWorker{louder, new Nan::Callback{callback}};
+    auto* worker = new AsyncHelloWorker{tile, new Nan::Callback{callback}};
```

Comment:
Now passing the [raw buffer](https://github.com/mapbox/vtvalidate/blob/f0a99d721865015fb4be532c72f6c586df45ab37/src/vtvalidate.cpp#L184)